### PR TITLE
Add GetVariableNoBarrier on brpc.

### PR DESCRIPTION
**Pull request by gongweibao**

### CODE REVIEW

File: paddle/fluid/operators/distributed/brpc/brpc_client.cc [hunk 1]
```diff
@@ -156,15 +156,18 @@ VarHandlePtr BRPCClient::_AsyncGetVar(const std::string& ep,
...
+                                      const std::string& out_var_name,
                                       const std::string& method_name,
                                       int64_t time_out) {
   const platform::DeviceContext* p_ctx = &ctx;
   const std::string ep_val = ep;
   const std::string var_name_val = var_name;
+  const std::string out_varname_val = out_var_name;
   const framework::Scope* p_scope = &scope;
   const auto ch_ptr = GetChannel(ep_val);
   const std::string method = "GetRPC";
```
Comment (seiriosPlus) [inline paddle/fluid/operators/distributed/brpc/brpc_client.cc]: can use k kSendRPC.

File: paddle/fluid/operators/distributed/brpc/brpc_server.cc [hunk 2]
```diff
@@ -122,23 +137,59 @@ class BRPCServiceImpl : public SendRecvService {
...
+                   "RequestGet handler should be registed first!");
+
+    brpc::ClosureGuard done_guard(done);
+    brpc::Controller* cntl = static_cast<brpc::Controller*>(cntl_butil);
+
+    std::string varname = request->varname();
+    std::string out_varname = request->out_varname();
     int trainer_id = request->trainer_id();
+
+    VLOG(3) << "RequestGet varname:" << varname
```
Comment (seiriosPlus) [inline paddle/fluid/operators/distributed/brpc/brpc_server.cc]: RequestGet -> RequestGetNoBarrier

File: paddle/fluid/operators/distributed/brpc/brpc_server.cc [hunk 3]
```diff
@@ -122,23 +137,59 @@ class BRPCServiceImpl : public SendRecvService {
...
+                                    &cntl->response_attachment(), "", false);
+    }
+  }
+
+  void _GetVariableNoBarrier(google::protobuf::RpcController* cntl_butil,
+                             const VariableMessage* request,
+                             VariableMessage* response,
+                             google::protobuf::Closure* done) {
+    PADDLE_ENFORCE(request_getnobarrier_h_ != nullptr,
+                   "RequestGet handler should be registed first!");
```
Comment (seiriosPlus) [inline paddle/fluid/operators/distributed/brpc/brpc_server.cc]: RequestGet -> RequestGetNoBarrier

File: paddle/fluid/operators/distributed/brpc/brpc_client.cc (hunk 1 above)
Comment (gongweibao) [reply paddle/fluid/operators/distributed/brpc/brpc_client.cc]: Done. Thanks!

File: paddle/fluid/operators/distributed/brpc/brpc_server.cc (hunk 3 above)
Comment (gongweibao) [reply paddle/fluid/operators/distributed/brpc/brpc_server.cc]: Done. Thanks!

File: paddle/fluid/operators/distributed/brpc/brpc_server.cc (hunk 2 above)
Comment (gongweibao) [reply paddle/fluid/operators/distributed/brpc/brpc_server.cc]: Done. Thanks!

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**